pocld/reply_th.cc

```diff
@@ -130,34 +131,27 @@ void ReplyQueueThread::pushReply(Reply *reply) {
   io_cond.notify_one();
 }
 
+void ReplyQueueThread::setConnection(
+    std::shared_ptr<Connection> NewConnection) {
+  std::unique_lock<std::mutex> l(ConnectionGuard);
+  Conn = NewConnection;
+  ConnectionNotifier.notify_one();
+}
+
 void ReplyQueueThread::writeThread() {
   // XXX: Change into a ring buffer?
   std::queue<Reply *> backup;
   bool resending = false;
   size_t i = 0;
-  int fd = *this->fd;
-  int oldfd = fd;
   while (1) {
   RETRY:
-    fd = *this->fd;
-    if (fd != oldfd) {
-      std::unique_lock<std::mutex> lock(io_mutex);
-      int n = io_inflight.size();
-      lock.unlock();
-      resending = true;
-      POCL_MSG_PRINT_GENERAL(
-          "%s: FD change detected with %d items in queue, %d -> %d\n",
-          id_str.c_str(), n, oldfd, fd);
-      // Reader closes the socket
-    }
-    oldfd = fd;
     if (eh->exit_requested())
       return;
 
     if (backup.empty())
       resending = false;
     std::unique_lock<std::mutex> lock(io_mutex);
-    if ((io_inflight.size() > 0 || resending) && fd >= 0) {
+    if ((io_inflight.size() > 0 || resending)) {
```

Comment:
```suggestion
    if ((IOInflight.size() > 0 || Resending)) {
```

Take the opportunity to C++ify the style (naming) here gradually.